src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -4190,8 +4193,11 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
       case PolyType(tparams, restpe) if tparams.nonEmpty =>
         if (sameLength(tparams, args)) {
           val targs = mapList(args)(_.tpe)
-          checkBounds(tree, NoPrefix, NoSymbol, tparams, targs, "")
-          if (fun.symbol.rawname == nme.classOf && currentRun.runDefinitions.isPredefClassOf(fun.symbol))
+          val isClassOf = fun.symbol.rawname == nme.classOf && currentRun.runDefinitions.isPredefClassOf(fun.symbol)
+          val skipBoundsCheck = isClassOf && context.unit.isJava
```

Comment:
I kind of don't like the duplicate and uncommented use of `context.unit.isJava`.

I remember when studying the differences in Akka's Java and Scala APIs that Java's `.class` and Scala's `classOf` have this non-trivial difference around raw/existential classes (_he says, with a hand-wavey "those things"_).  So, I wonder: what do you think if Java's `List.class` were a new thing, a variant of `nme.classOf`?  Obviously, it would be a more invasive change, but I feel it would be better than trying to spot the missing `if` condition and throwing enough variant Java code at the compiler.

Reply:
I've pushed an alternative approach that nips the raw type in the bud.